### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by alexcasalboni**
This PR adds a few more unit tests to increase overall coverage and potentially identify bugs/errors.

It definitely needs some more word to cover more edge cases and validate the results (for now it's only focused on reaching decent coverage).

```
Filename                      Regions    Missed Regions     Cover   Functions  Missed Functions  Executed       Lines      Missed Lines     Cover    Branches   Missed Branches     Cover
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
ga_payload.rs                     126                21    83.33%          16                 0   100.00%         246                 2    99.19%           0                 0         -
lib.rs                            295                19    93.56%          33                 5    84.85%         502                18    96.41%           0                 0         -
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
TOTAL                             421                40    90.50%          49                 5    89.80%         748                20    97.33%           0                 0         -

```

### CODE REVIEW

**Comment by SachaMorard** [comment]
@alexcasalboni wow, it's really cool. LGTM

**Comment by alexcasalboni** [comment]
@SachaMorard I've done a minor cleanup, updated the README, and added coverage to the GitHub Action.

I'll wait for some feedback form @KokaKiwi and @CLEMENTINATOR, then eventually we can reuse this as the baseline for all the other data collection components (since only 1 test is GA-specific).

If you're ok with it, I would also add a nice coverage badge on the README (via Coveralls or similar).

File: src/lib.rs [hunk 1]
```diff
@@ -347,4 +351,354 @@ mod tests {
...
+            Some(Consent::Granted),
+            "abc".to_string(),
+            "fr".to_string(),
+            true,
+        );
+        let credentials = sample_credentials();
+        let result = GaComponent::page(event, credentials);
+
+        assert_eq!(result.is_err(), false);
+        let edgee_request = result.unwrap();
```
Comment (KokaKiwi) [inline src/lib.rs:540]: Maybe using `?` to be more concise (or not, idk actually) would be useful: https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#tests-and-

(It's no problem unwrapping in test, just got reminded that we can return a `Result` in tests too :smile:)

**Review by KokaKiwi — APPROVED** [review]

**Comment by alexcasalboni** [comment]
Sorry for the post-approval updates :) nothing major, just a tiny refactor to use the same test functions in other components and I've updated the `dev-dependencies` in the cargo.toml file for easier local setup.

I'll merge now 🚀
